Client/game_sa/CFxSA.cpp

```diff
@@ -11,6 +11,25 @@
 
 #include "StdInc.h"
 
+using StoreShadowToBeRendered_t = int(__cdecl*)(char type, RwTexture* texture, CVector* pos, float x1, float y1, float x2, float y2, __int16 intensity, char r,
+                                                char g, char b, float zDistance, char bDrawOnWater, float scale, void* shadowData, char bDrawOnBuildings);
+auto StoreShadowToBeRendered = (StoreShadowToBeRendered_t)0x707390;
+
+CFxSA::CFxSA(CFxSAInterface* pInterface)
+{
+    m_pInterface = pInterface;
+
+    RwTexture* pRwTexture;
+    for (int i = 0; i < SHADOW_COUNT; i++)
+    {
+        void* textureAddress;
+        memcpy(&textureAddress, (void*)(SHADOW_BASE_TEXTURE_OFFSET + i * 4), 4);
+        pRwTexture = reinterpret_cast<RwTexture*>((void*)textureAddress);
+        m_textureMap[(eShadowType)i] = pRwTexture;
```

Comment:
```suggestion
        m_textureMap[static_cast<eShadowType>(i)] = pRwTexture;
```

(Source https://stackoverflow.com/a/53148670/1517394)